launcher/src/se/llbit/chunky/launcher/ChunkyLauncher.java

```diff
@@ -61,150 +61,176 @@ protected static void launchFailure(String command) {
+                  ChunkyLauncher::launchFailure, loggerBuilder);
+          if(exitCode != 0) {
+            System.exit(exitCode);
+          }
+        } else {
+          System.err.println("Could not launch selected Chunky version. Try updating with --update");
+          System.exit(1);
         }
       } else {
-        System.err.println("Could not launch selected Chunky version. Try updating with --update");
-        System.exit(1);
-      }
-    } else {
-      final boolean finalForceLauncher = forceLauncher;
-      // A callback is used to decide if the launcher should be displayed after
-      // the first time setup dialog has been shown (if needed).
-      firstTimeSetup(settings, () -> {
-        ChunkyDeployer.deploy(settings); // Install the embedded version.
+        final boolean finalForceLauncher = forceLauncher;
+        // A callback is used to decide if the launcher should be displayed after
+        // the first time setup dialog has been shown (if needed).
+        firstTimeSetup(settings, () -> {
+          ChunkyDeployer.deploy(settings); // Install the embedded version.
 
-        if (!finalForceLauncher && !settings.showLauncher) {
-          // Skip the launcher only if we can launch this version.
-          VersionInfo version = ChunkyDeployer.resolveVersion(settings.version);
-          if (ChunkyDeployer.canLaunch(version, null, false)) {
-            if (ChunkyDeployer.launchChunky(settings, version, LaunchMode.GUI,
-                ChunkyLauncher::launchFailure,
-                loggerBuilder) == 0) {
-              return false;
+          if(!finalForceLauncher && !settings.showLauncher) {
+            // Skip the launcher only if we can launch this version.
+            VersionInfo version = ChunkyDeployer.resolveVersion(settings.version);
+            if(ChunkyDeployer.canLaunch(version, null, false)) {
+              if(ChunkyDeployer.launchChunky(settings, version, LaunchMode.GUI,
+                      ChunkyLauncher::launchFailure,
+                      loggerBuilder) == 0) {
+                return false;
+              }
             }
           }
-        }
-        return true;
-      });
+          return true;
+        });
+      }
+    } catch(NoClassDefFoundError e) {
+      String cause = e.getMessage();
+      if(cause.contains("javafx")) {
+        // Javafx error
+        if(retryIfMissingJavafx)
+          JavaFxLocator.retryWithJavafx(args);
+        System.err.println("Error: Java cannot find javafx.");
+        System.err.println("If you are using a JVM for java 11 or later, " +
+                "javafx is no longer shipped alongside and must be installed separately");
+        System.err.println("If you already have javafx installed, you need to run chunky with the command:");
+        System.err.println("java --module-path <path/to/javafx/lib> --add-modules javafx.controls,javafx.fxml -jar <path/to/ChunkyLauncher.jar>");
```

Comment:
Can this message be displayed in some dialog so that it also pops up when not launching Chunky from CLI?

Reply:
Probably using Swing?

Reply:
`JOptionPane.showMessageDialog(null, "Oh no!")`?